src/simulations/collect.jl

```diff
@@ -0,0 +1,338 @@
+###################################################
+# Definition of the data collection API
+###################################################
+get_data(a, s::Symbol) = getproperty(a, s)
+get_data(a, f::Function) = f(a)
+
+# TODO: `add_data` is a misleading name, we need a better one.
+add_data(s, model, when::AbstractVector) = s ∈ when
```

Comment:
`inWhen`?

Reply:
`inwhen` is good, and I was also thinking something crazy verbose like `should_we_collect` :D If julia allowed `?` at the end of function names iot would be super cool :D

Reply:
yeah. Or `time_to_collect`